Fix compilation issue on Solaris 11.4:

../deps/v8/src/base/platform/platform-posix.cc:97:16: error: conflicting declaration of C function 'int madvise(caddr_t, std::size_t, int)'
   97 | extern "C" int madvise(caddr_t, size_t, int);
      |                ^~~~~~~
In file included from ../deps/v8/src/base/platform/platform-posix.cc:20:
/usr/include/sys/mman.h:242:12: note: previous declaration 'int madvise(void*, std::size_t, int)'
  242 | extern int madvise(void *, size_t, int);
      |            ^~~~~~~

--- node-22.22.0/deps/v8/src/base/platform/platform-posix.cc
+++ node-22.22.0/deps/v8/src/base/platform/platform-posix.cc
@@ -76,26 +76,6 @@ #if V8_OS_FREEBSD || V8_OS_DARWIN || V8_
 #define MAP_ANONYMOUS MAP_ANON
 #endif
 
-/*
- * NOTE: illumos starting with illumos#14418 (pushed April 20th, 2022)
- * prototypes madvise(3C) properly with a `void *` first argument.
- * The only way to detect this outside of configure-time checking is to
- * check for the existence of MEMCNTL_SHARED, which gets defined for the first
- * time in illumos#14418 under the same circumstances save _STRICT_POSIX, which
- * thankfully neither Solaris nor illumos builds of Node or V8 do.
- *
- * If some future illumos push changes the MEMCNTL_SHARED assumptions made
- * above, the illumos check below will have to be revisited.  This check
- * will work on both pre-and-post illumos#14418 illumos environments.
- */
-#if defined(V8_OS_SOLARIS) && !(defined(__illumos__) && defined(MEMCNTL_SHARED))
-#if (defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE > 2) || defined(__EXTENSIONS__)
-extern "C" int madvise(caddr_t, size_t, int);
-#else
-extern int madvise(caddr_t, size_t, int);
-#endif
-#endif
-
 #ifndef MADV_FREE
 #define MADV_FREE MADV_DONTNEED
 #endif
